docs/domains/sparse_linear_algebra.rst

```diff
@@ -38,3 +38,122 @@ Currently known limitations:
   ``oneapi::mkl::unimplemented`` exception.
 - Scalar parameters ``alpha`` and ``beta`` should be host pointers to prevent
   synchronizations and copies to the host.
+
+
+cuSPARSE backend
+----------------
+
+Currently known limitations:
+
+- The COO format requires the indices to be sorted by row. See the `cuSPARSE
+  documentation
+  <https://docs.nvidia.com/cuda/cusparse/index.html#coordinate-coo>`_. Sparse
+  operations using matrices with the COO format without the property
+  ``matrix_property::sorted_by_rows`` or ``matrix_property::sorted`` will throw
+  an ``oneapi::mkl::unimplemented`` exception.
+- Using ``spmm`` with the algorithm ``spmm_alg::csr_alg3`` and an ``opA`` other
+  than ``transpose::nontrans`` or an ``opB`` ``transpose::conjtrans`` will throw
+  an ``oneapi::mkl::unimplemented`` exception.
+- Using ``spmv`` with a ``type_view`` other than ``matrix_descr::general`` will
+  throw an ``oneapi::mkl::unimplemented`` exception.
+
+
+Operation algorithms mapping
+----------------------------
+
+The following tables describe how a oneMKL SYCL Interface algorithm maps to the
+backend's algorithms. Refer to the backend's documentation for a more detailed
+explanation of the algorithms.
+
+Backends with no equivalent algorithms will fallback to the backend's default
+behavior.
```

Comment:
We actually have a tricky situation ahead of us now with these algorithms and optional calls like cuSPARSE's [`cusparseSpMM_preprocess`](https://docs.nvidia.com/cuda/cusparse/#cusparsespmm).
Because their preprocess call is optional, has no descriptor, and takes in an external workspace that is also passed again in the `spmm` API, preprocess() calls may or may not work with multiple algorithm enums. (effectively up-to-doubling of the number of enums available). With the APIs we have currently, we can't support that. Our `default_alg` calls preprocess+spmm calls, whereas `no_optimize_alg` enum is `default_alg` the preprocess call.

Their doc says:
> The function `cusparseSpMM_preprocess()` can be called before `cusparseSpMM` to speedup the actual computation. It is useful when `cusparseSpMM` is called multiple times with the same sparsity pattern (`matA`). The values of the matrices (`matA`, `matB`, `matC`) can change arbitrarily. **It provides performance advantages is used with `CUSPARSE_SPMM_CSR_ALG1` or `CUSPARSE_SPMM_CSR_ALG3`. For all other formats and algorithms have no effect.**

This means that for cuSPARSE backend, for `CUSPARSE_SPMM_CSR_ALG1` and `CUSPARSE_SPMM_CSR_ALG3`, we do not support calling `spmm` without `preprocess` whereas cuSPARSE supports it.

I'm okay with this missing support on our/Spec side at least for now until someone complains about it some day.

Reply:
Maybe we should add a documentation note about this here or somewhere else?

Reply:
That is a very good point! This is now documented in https://github.com/oneapi-src/oneMKL/pull/527/commits/5addc1479936302a3624a8aa3b95e95234ed78a1